subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/MoreMemorySampleModifier.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.fixtures.executer;
+
+import org.gradle.samples.model.Command;
+import org.gradle.samples.model.Sample;
+import org.gradle.samples.test.runner.SampleModifier;
+import org.gradle.util.GUtil;
+
+import java.io.File;
+import java.util.Properties;
+
+/**
+ * Gives the sample tests double the normal memory, as they seem to be very resource intensive.
+ *
+ * //TODO find out why these tests need so much memory and fix it.
+ */
+public class MoreMemorySampleModifier implements SampleModifier {
+
+    private static final String ORG_GRADLE_JVMARGS = "org.gradle.jvmargs";
+
+    @Override
+    public Sample modify(Sample sample) {
+        File projectDir = sample.getProjectDir();
+        giveMoreMemoryTo(projectDir);
+        for (Command command : sample.getCommands()) {
+            if (command.getExecutionSubdirectory() != null) {
+                giveMoreMemoryTo(new File(projectDir, command.getExecutionSubdirectory()));
+            }
+        }
+        return sample;
+    }
+
+    private void giveMoreMemoryTo(File projectDir) {
+        File propertiesFile = new File(projectDir, "gradle.properties");
+        Properties properties = propertiesFile.exists() ? GUtil.loadProperties(propertiesFile) : new Properties();
+        String existingArgs = properties.getProperty(ORG_GRADLE_JVMARGS, "");
+        properties.setProperty(ORG_GRADLE_JVMARGS, "-Xmx1024m -XX:MaxMetaspaceSize=512m -XX:+HeapDumpOnOutOfMemoryError " + existingArgs);
```

Comment:
This is a little odd - Our sample tests should have failed on Java 7 before if they can't deal with 256m of space for classes. The only reason I can come up with is that we switched to Exemplar *after* we removed Java 7 coverage.

In any case, we should find out why they take so much metaspace. They aren't very big or complex, so our normal threshold should be enough.